server/src/game/role/marksman.rs

```diff
@@ -0,0 +1,208 @@
+        *marks = match marks {
+            MarksmanMarks::None => {
+                MarksmanMarks::One { a: p }
+            }
+            MarksmanMarks::One {a} => {
+                if *a != p {
+                    MarksmanMarks::Two { a: *a, b: p }
+                }else{
+                    MarksmanMarks::None
+                }
+            }
+            MarksmanMarks::Two { a, b } => {
+                if p == *a {
+                    MarksmanMarks::One { a: *b }
+                }else if p == *b {
+                    MarksmanMarks::One { a: *a }
+                }else{
+                    MarksmanMarks::Two { a: *a, b: *b }
+                }
+            }
+        };
+    }
+}
+#[derive(Clone, Debug, Serialize, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+#[serde(tag = "type")]
+enum MarksmanMarks{
+    None,
+    One{
+        a: PlayerReference
+    },
+    Two{
+        a: PlayerReference, 
+        b: PlayerReference
+    }
+}
+impl MarksmanMarks{
+    fn marks(&self)->Box<[PlayerReference]> {
+        match self {
+            MarksmanMarks::None => {Box::new([])},
+            MarksmanMarks::One{a} => {Box::new([*a])},
+            MarksmanMarks::Two{a,b} => {Box::new([*a,*b])}
+        }
+    }
+}
+
+impl Default for Marksman {
+    fn default() -> Self {
+        Self { state: MarksmanState::NotLoaded }
+    }
+}
+
+pub(super) const FACTION: Faction = Faction::Town;
+pub(super) const MAXIMUM_COUNT: Option<u8> = None;
+
+impl RoleStateImpl for Marksman {
+    fn defense(&self, _game: &Game, _actor_ref: PlayerReference) -> u8 {0}
+    
+
+    fn do_night_action(mut self, game: &mut Game, actor_ref: PlayerReference, priority: Priority) {
+        
+    
+        match priority{
+            Priority::Kill => {
+                let visiting_players: Vec<_> = actor_ref
+                    .night_visits(game)
+                    .into_iter()
+                    .flat_map(|p|p.target.all_visitors(game))
+                    .collect();
+
+                for mark in self.state.marks().into_iter() {
+                    
+                    if !visiting_players.contains(&mark) {continue};
+                    
+                    actor_ref.push_night_message(game, ChatMessageVariant::MarksmanAttackedVisitor);
+                    let killed = mark.try_night_kill(actor_ref, game, GraveKiller::Role(Role::Marksman), 1, false);
+
+                    if killed && mark.role(game).faction() == Faction::Town {
+                        self.state = MarksmanState::ShotTownie;
+                    }
+                }
+                
+                actor_ref.set_role_state(game, RoleState::Marksman(self));
+            },
+            _ => {}
+        }
+
+    }
+    fn do_day_action(mut self, game: &mut Game, actor_ref: PlayerReference, target_ref: PlayerReference) {
+        self.state.toggle_mark(target_ref);
+        if self.state.marks().len() == 0 {
+            actor_ref.set_chosen_targets(game, vec![]);
```

Comment:
Seems weird. You probably don't want this

Reply:
??? Im resolving this and ignoring it